diff --git a/lib/notifiers/item_notifier.dart b/lib/notifiers/item_notifier.dart
index 378ebe3..03b132a 100644
--- a/lib/notifiers/item_notifier.dart
+++ b/lib/notifiers/item_notifier.dart
@@ -122,6 +122,26 @@ class ItemNotifier extends StateNotifier<List<ListItem>> {
     // state = [];
   }
 
+  Future<void> deleteCollected() async {
+    final uid = FirebaseAuth.instance.currentUser!.uid;
+    final snapshot = await FirebaseFirestore.instance
+        .collection('listItems')
+        .where('userId', isEqualTo: uid)
+        .where('acquired', isEqualTo: true)
+        .get();
+
+    // Use a batch for efficient deletion
+    final batch = FirebaseFirestore.instance.batch();
+    for (DocumentSnapshot ds in snapshot.docs) {
+      batch.delete(ds.reference);
+    }
+
+    // Commit the batch
+    await batch.commit();
+
+    // state = [];
+  }
+
   Future<void> toggleAcquiredProvider(ListItem item) async {
     if (item.id.isNotEmpty) {
       final newValue = item.toggleAcquired();
diff --git a/lib/screens/items_screen.dart b/lib/screens/items_screen.dart
index e3efec3..1231cb3 100644
--- a/lib/screens/items_screen.dart
+++ b/lib/screens/items_screen.dart
@@ -1,18 +1,18 @@
 import 'package:firebase_auth/firebase_auth.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:flutter_typeahead/flutter_typeahead.dart';
 import 'package:shopping_list/models/list_item.dart';
 import 'package:shopping_list/notifiers/item_notifier.dart';
 import 'package:shopping_list/utils/categorize_items.dart';
 import 'package:shopping_list/utils/get_suggestions.dart';
 import 'package:shopping_list/widgets/about_content.dart';
-import 'package:flutter_typeahead/flutter_typeahead.dart';
 
 class ItemsScreen extends ConsumerStatefulWidget {
-  const ItemsScreen({required this.toHomeScreen, super.key});
-
   final void Function() toHomeScreen;
 
+  const ItemsScreen({required this.toHomeScreen, super.key});
+
   @override
   ConsumerState<ItemsScreen> createState() {
     return _ItemsScreenState();
@@ -25,58 +25,6 @@ class _ItemsScreenState extends ConsumerState<ItemsScreen> {
   final GlobalKey<FormState> _keyDialogForm = GlobalKey<FormState>();
   bool isCategorizing = false;
 
-  @override
-  void dispose() {
-    itemController.dispose();
-    amountController.dispose();
-    super.dispose();
-  }
-
-  Future<void> showDeleteAllDialog() async {
-    return showDialog<void>(
-        context: context,
-        builder: (BuildContext context) {
-          return AlertDialog(
-            title: const Text('Delete all?'),
-            content: const Text(
-                'If you continue, all items from the list will disappear.'),
-            actions: <Widget>[
-              TextButton(
-                  onPressed: () async {
-                    // await deleteAll();
-                    ref.read(itemProvider.notifier).deleteAll();
-                    if (!mounted) return;
-                    Navigator.of(context).pop();
-                  },
-                  child: const Text('Delete')),
-              TextButton(
-                  onPressed: () {
-                    Navigator.of(context).pop();
-                  },
-                  child: const Text('Cancel'))
-            ],
-          );
-        });
-  }
-
-  Future<void> showAbout() async {
-    return showDialog(
-        context: context,
-        builder: (BuildContext context) {
-          return AlertDialog(
-            title: const Text('About'),
-            content: const AboutContent(),
-            actions: <Widget>[
-              TextButton(
-                  onPressed: () {
-                    Navigator.of(context).pop();
-                  },
-                  child: const Text('Back'))
-            ],
-          );
-        });
-  }
-
   @override
   Widget build(BuildContext context) {
     //final Future<List<ListItem>> _listItems = fetchItems();
@@ -210,6 +158,7 @@ class _ItemsScreenState extends ConsumerState<ItemsScreen> {
                     ),
         ),
         bottomNavigationBar: BottomAppBar(
+          color: Colors.white,
           padding: EdgeInsets.all(0),
           child: Padding(
             padding: EdgeInsets.zero,
@@ -222,13 +171,14 @@ class _ItemsScreenState extends ConsumerState<ItemsScreen> {
                     mainAxisSize: MainAxisSize.min,
                     children: [
                       IconButton(
+                        color: Colors.red,
                         onPressed: () async {
-                          await showDeleteAllDialog();
+                          await _showDeleteOptions(context);
                         },
                         icon: Icon(Icons.delete),
                       ),
                       Text(
-                        'Delete all',
+                        'Delete…',
                         style: Theme.of(context).textTheme.labelSmall,
                       ),
                     ],
@@ -238,6 +188,7 @@ class _ItemsScreenState extends ConsumerState<ItemsScreen> {
                   mainAxisSize: MainAxisSize.min,
                   children: [
                     IconButton(
+                      color: Colors.green,
                       icon: Icon(Icons.category),
                       onPressed: () async {
                         setState(() {
@@ -276,6 +227,7 @@ class _ItemsScreenState extends ConsumerState<ItemsScreen> {
                   mainAxisSize: MainAxisSize.min,
                   children: [
                     IconButton(
+                      color: Colors.blue,
                       icon: const Icon(Icons.info),
                       onPressed: showAbout,
                     ),
@@ -289,6 +241,7 @@ class _ItemsScreenState extends ConsumerState<ItemsScreen> {
                   mainAxisSize: MainAxisSize.min,
                   children: [
                     IconButton(
+                      color: Colors.brown,
                       icon: const Icon(Icons.exit_to_app),
                       onPressed: () async {
                         await FirebaseAuth.instance.signOut();
@@ -309,6 +262,31 @@ class _ItemsScreenState extends ConsumerState<ItemsScreen> {
         ));
   }
 
+  @override
+  void dispose() {
+    itemController.dispose();
+    amountController.dispose();
+    super.dispose();
+  }
+
+  Future<void> showAbout() async {
+    return showDialog(
+        context: context,
+        builder: (BuildContext context) {
+          return AlertDialog(
+            title: const Text('About'),
+            content: const AboutContent(),
+            actions: <Widget>[
+              TextButton(
+                  onPressed: () {
+                    Navigator.of(context).pop();
+                  },
+                  child: const Text('Back'))
+            ],
+          );
+        });
+  }
+
   Future showaddItemDialog({List<ListItem>? listItems, ListItem? item}) async {
     final newItem = item == null;
     bool editItem = false;
@@ -458,4 +436,67 @@ class _ItemsScreenState extends ConsumerState<ItemsScreen> {
           );
         });
   }
+
+  Future<void> _showDeleteAllDialog() async {
+    return showDialog<void>(
+        context: context,
+        builder: (BuildContext context) {
+          return AlertDialog(
+            title: const Text('Delete all?'),
+            content: const Text(
+                'If you continue, all items from the list will disappear.'),
+            actions: <Widget>[
+              TextButton(
+                  onPressed: () async {
+                    // await deleteAll();
+                    ref.read(itemProvider.notifier).deleteAll();
+                    if (!mounted) return;
+                    Navigator.of(context).pop();
+                  },
+                  child: const Text('Delete')),
+              TextButton(
+                  onPressed: () {
+                    Navigator.of(context).pop();
+                  },
+                  child: const Text('Cancel'))
+            ],
+          );
+        });
+  }
+
+  Future<void> _showDeleteOptions(BuildContext context) async {
+    showModalBottomSheet(
+      context: context,
+      shape: RoundedRectangleBorder(
+        borderRadius: BorderRadius.vertical(top: Radius.circular(16)),
+      ),
+      builder: (context) {
+        return Wrap(
+          children: [
+            ListTile(
+              leading: Icon(Icons.delete_forever, color: Colors.red),
+              title: Text('Delete All'),
+              onTap: () async {
+                await _showDeleteAllDialog(); // Call delete all function
+                if (!mounted) return;
+                Navigator.pop(context);
+              },
+            ),
+            ListTile(
+              leading: Icon(Icons.delete_sweep, color: Colors.orange),
+              title: Text('Delete Collected'),
+              onTap: () async {
+                ref.read(itemProvider.notifier).deleteCollected();
+
+                Navigator.pop(context);
+              },
+            ),
+            const SizedBox(
+              height: 5,
+            )
+          ],
+        );
+      },
+    );
+  }
 }
diff --git a/pubspec.yaml b/pubspec.yaml
index 408050e..526bfb7 100644
--- a/pubspec.yaml
+++ b/pubspec.yaml
@@ -16,7 +16,7 @@ publish_to: 'none' # Remove this line if you wish to publish to pub.dev
 # https://developer.apple.com/library/archive/documentation/General/Reference/InfoPlistKeyReference/Articles/CoreFoundationKeys.html
 # In Windows, build-name is used as the major, minor, and patch parts
 # of the product and file versions while build-number is used as the build suffix.
-version: 1.0.1+14
+version: 1.0.1+15
 
 environment:
   sdk: '>=3.4.0 <4.0.0'
